release/src/router/www/Main_IPV6Status_Content.asp

```diff
@@ -132,36 +132,43 @@ function show_pinholes() {
 	var code, i, line;
 
 	code = '<table width="100%" border="1" align="center" cellpadding="4" cellspacing="0" bordercolor="#6b8fa3"class="FormTable_table">';
-	code += '<thead><tr><td colspan="5">IPv6 firewall Pinholes</td></tr></thead>';
-	code += '<tr><th width="10%">Proto</th>';
-	code += '<th width="35%">Remote IP</th>';
-	code += '<th width="10%">Port</th>';
-	code += '<th width="35%">Local IP</th>';
-	code += '<th width="10%">Port</th>';
+	code += '<thead><tr><td colspan="6">UPnP IGDv2 &amp; PCP IPv6 port forwards</td></tr></thead>';
+	code += '<tr><th width="6%">Proto.</th>';
+	code += '<th width="32%">Remote IP</th>';
+	code += '<th width="9%">Rem. Port</th>';
+	code += '<th width="32%">Local IP</th>';
+	code += '<th width="9%">Local Port</th>';
+	code += '<th width="12%">Time left</th>';
 	code += '</tr>';
 
-	if ("<% nvram_get("upnp_pinhole_enable"); %>" == "0") {
-		code += '<tr><td colspan="5">Pinhole support is currently disabled.</td></tr>';
+	if ("<% nvram_get("wan_upnp_enable"); %>" == "0") {
```

Comment:
Checking for UPnP being enabled is not enough, upnp_pinhole_enable must also be set to 1 for pinholes to be enabled.  Both vars need to be checked.

Reply:
It shouldn't.  The point of this setting is to specifically prevent clients from opening IPv6 pinhole in the firewall (hence its name), in addition to increasing backward compatibility.  The webui description also explicitely calls it "[Enable UPnP IGDv2 (IPv6 support)]".